### ISSUE TITLE
TestLinearDensityProbe does not build

### ISSUE THREAD

**Issue by nsoblath** [project8/katydid#81]
Building on a recent branch off of develop on a Mac.

```
[ 84%] Building CXX object Source/Executables/Validation/CMakeFiles/TestLinearDensityProbe.dir/TestLinearDensityProbe.cc.o
/Users/obla999/Work/Project8/Software/katydid/Source/Executables/Validation/TestLinearDensityProbe.cc:140:16: error: no member named 'SetStepSizeBig' in
      'Katydid::KTLinearDensityProbeFit'
    lineFitter.SetStepSizeBig( 0.2e6 );
    ~~~~~~~~~~ ^
/Users/obla999/Work/Project8/Software/katydid/Source/Executables/Validation/TestLinearDensityProbe.cc:141:16: error: no member named 'SetStepSizeSmall' in
      'Katydid::KTLinearDensityProbeFit'
    lineFitter.SetStepSizeSmall( 0.004e6 );
    ~~~~~~~~~~ ^
/Users/obla999/Work/Project8/Software/katydid/Source/Executables/Validation/TestLinearDensityProbe.cc:145:21: error: no member named 'Calculate' in 'Katydid::KTLinearDensityProbeFit'
    if( !lineFitter.Calculate( tr, threshPts ) )
         ~~~~~~~~~~ ^
In file included from /Users/obla999/Work/Project8/Software/katydid/Source/Executables/Validation/TestLinearDensityProbe.cc:8:
In file included from /Users/obla999/Work/Project8/Software/katydid/Source/EventAnalysis/KTLinearDensityProbeFit.hh:12:
In file included from /Users/obla999/Work/Project8/Software/katydid/Source/Data/SpectrumAnalysis/KTGainVariationData.hh:11:
In file included from /Users/obla999/Work/Project8/Software/katydid/Nymph/Library/Data/KTData.hh:11:
/Users/obla999/Work/Project8/Software/katydid/Nymph/Library/Utility/KTExtensibleStruct.hh:163:13: error: 'const Katydid::KTPSCollectionData' is an incomplete type
        if (dynamic_cast<const XStructType*>(this))
            ^           ~~~~~~~~~~~~~~~~~~~~
/Users/obla999/Work/Project8/Software/katydid/Source/EventAnalysis/KTLinearDensityProbeFit.hh:290:21: note: in instantiation of function template specialization
      'Nymph::KTExtensibleStructCore<Nymph::KTDataCore>::Has<Katydid::KTPSCollectionData>' requested here
        if (! data->Has< KTPSCollectionData >())
                    ^
/Users/obla999/Work/Project8/Software/katydid/Source/EventAnalysis/KTLinearDensityProbeFit.hh:28:11: note: forward declaration of 'Katydid::KTPSCollectionData'
    class KTPSCollectionData;
          ^
In file included from /Users/obla999/Work/Project8/Software/katydid/Source/Executables/Validation/TestLinearDensityProbe.cc:8:
In file included from /Users/obla999/Work/Project8/Software/katydid/Source/EventAnalysis/KTLinearDensityProbeFit.hh:12:
In file included from /Users/obla999/Work/Project8/Software/katydid/Source/Data/SpectrumAnalysis/KTGainVariationData.hh:11:
In file included from /Users/obla999/Work/Project8/Software/katydid/Nymph/Library/Data/KTData.hh:11:
/Users/obla999/Work/Project8/Software/katydid/Nymph/Library/Utility/KTExtensibleStruct.hh:123:31: error: 'Katydid::KTPSCollectionData' is an incomplete type
        XStructType* target = dynamic_cast<XStructType*>(this);
                              ^           ~~~~~~~~~~~~~~
/Users/obla999/Work/Project8/Software/katydid/Source/EventAnalysis/KTLinearDensityProbeFit.hh:297:116: note: in instantiation of function template specialization
      'Nymph::KTExtensibleStructCore<Nymph::KTDataCore>::Of<Katydid::KTPSCollectionData>' requested here
        if( !ChooseAlgorithm( data->Of< KTProcessedTrackData >(), data->Of< KTDiscriminatedPoints2DData >(), data->Of< KTPSCollectionData >() ) )
                                                                                                                   ^
/Users/obla999/Work/Project8/Software/katydid/Source/EventAnalysis/KTLinearDensityProbeFit.hh:28:11: note: forward declaration of 'Katydid::KTPSCollectionData'
    class KTPSCollectionData;
          ^
In file included from /Users/obla999/Work/Project8/Software/katydid/Source/Executables/Validation/TestLinearDensityProbe.cc:8:
In file included from /Users/obla999/Work/Project8/Software/katydid/Source/EventAnalysis/KTLinearDensityProbeFit.hh:12:
In file included from /Users/obla999/Work/Project8/Software/katydid/Source/Data/SpectrumAnalysis/KTGainVariationData.hh:11:
In file included from /Users/obla999/Work/Project8/Software/katydid/Nymph/Library/Data/KTData.hh:11:
/Users/obla999/Work/Project8/Software/katydid/Nymph/Library/Utility/KTExtensibleStruct.hh:131:25: error: allocation of incomplete type 'Katydid::KTPSCollectionData'
            fNext = new XStructType();
                        ^~~~~~~~~~~
/Users/obla999/Work/Project8/Software/katydid/Source/EventAnalysis/KTLinearDensityProbeFit.hh:28:11: note: forward declaration of 'Katydid::KTPSCollectionData'
    class KTPSCollectionData;
          ^
6 errors generated.
make[2]: *** [Source/Executables/Validation/CMakeFiles/TestLinearDensityProbe.dir/TestLinearDensityProbe.cc.o] Error 1
make[1]: *** [Source/Executables/Validation/CMakeFiles/TestLinearDensityProbe.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by evzayas** [comment]
ah, I haven't kept the validation up to date with the rotate-and-project additions to this class. I will add this to my to-do list tomorrow.

**Comment by nsoblath** [comment]
Fixed by @evzayas in commit 59585f82207800f10464eb9057c8f0784b213414
